tests/test_utils/test_testcase.py

```diff
@@ -65,14 +65,14 @@ def test_disallowed_database_queries(self):
 
     @skipUnlessDBFeature("supports_transactions")
     def test_reset_sequences(self):
-        old_reset_sequences = self.reset_sequences
-        self.reset_sequences = True
+        old_reset_sequences = self.__class__.reset_sequences
+        self.__class__.reset_sequences = True
         msg = "reset_sequences cannot be used on TestCase instances"
         try:
```

Comment:
Since we are changing this, could we instead use `addCleanup` and remove the `try-finally`:
```suggestion
        self.addCleanup(setattr, self.__class__, reset_sequences, old_reset_sequences)
```